---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```
# GECKO

Accurate genetic and environmental covariance estimation with composite likelihood in genome-wide association studies

## Installation
To install the GECKO, it's easiest to use the 'devtools' package.

``` r
#install.packages("devtools")
library(devtools)
install_github("borangao/GECKO")
```

## Usage

The following help page will provide quick references for GECKO package and the example command lines:
```r 
library(GECKO)
package?GECKO
###Read in the summary statistics
data(sumstat_1)
data(sumstat_2)
data(ldscore)
####sumstat_1 and sumstat_2 are the files of the summary statistics with column name chr, bp, SNP, A1, A2, N, Z, P representing chromosome, base pair position, SNP iD, major allele, minor allele, number of individuals in the study, Z score, P value
###ldscore is the LD score file generated by LDSC software using 1000 genome reference panel
n1in<-round(mean(sumstat_1$N))
n2in<-round(mean(sumstat_2$N))
nsin<-0
Weightin = T #is always set for GECKO to improve the efficiency
Fix_Vein = T #(if two studies have non-overlapped samples, otherwise false)
Test = T #Test for significance or not
###Need to specify the number of individuals within each study: n1,n2, and number of the overlapping individuals in the two studies:nsin
###if the two samples are from the separate studies, nsin = 0, and Fix_Vein = 1
###if the two samples are from the same study, nsin need to be specified

Result<-GECKO_R(sumstat_1,sumstat_2,n1in,n2in,nsin,ldscore,Weightin,Fix_Vein,Test)
### Use GECKO_R function to calculate genetic and environmental covariance
```

## Development

This package is developed and maintained by Boran Gao (borang@umich.edu).




