
<!-- README.md is generated from README.Rmd. Please edit that file -->
GECKO
=====

Accurate genetic and environmental covariance estimation with composite likelihood in genome-wide association studies

Installation
------------

To install the GECKO, it's easiest to use the 'devtools' package.

``` r
#install.packages("devtools")
library(devtools)
install_github("borangao/GECKO")
```

Usage
-----

The following help page will provide quick references for GECKO package, an example data is provided with the package.

``` r
library(GECKO)
package?GECKO
```

### Access example data

sumstat\_1 and sumstat\_2 are the files of the summary statistics with column name chr, bp, SNP, A1, A2, N, Z, P representing chromosome, base pair position, SNP iD, major allele, minor allele, number of individuals in the study, Z score, P value. ldscore is the LD score file generated by LDSC software using 1000 genome reference panel

``` r
data(sumstat_1)
data(sumstat_2)
data(ldscore)
```

### Analysis Code

There are 6 arguments needed to be specified by GECKO including number of observation of the both studies, number of overlapped samples, whether to use weighted composite-lieklihood (usually is specified TRUE to increase estimation accuracy), whether fix environmental covariance to be zero if it's known that there is no overlapped sample, whether testing for the genetic and envrionmental covariance (It takes more time to test for the significant genetic and envrionmental covariance).GECKO\_R is the function to calculate genetic and environmental covariance

``` r
n1in<-round(mean(sumstat_1$N))
n2in<-round(mean(sumstat_2$N))
nsin<-0
Weightin = T #is always set for GECKO to improve the efficiency
Fix_Vein = T #(if two studies have non-overlapped samples, otherwise false)
Test = T #Test for significance or not
###Need to specify the number of individuals within each study: n1,n2, and number of the overlapping individuals in the two studies:nsin
###if the two samples are from the separate studies, nsin = 0, and Fix_Vein = 1
###if the two samples are from the same study, nsin need to be specified

Result<-GECKO_R(sumstat_1,sumstat_2,n1in,n2in,nsin,ldscore,Weightin,Fix_Vein,Test)
```

Development
-----------

This package is developed and maintained by Boran Gao (<borang@umich.edu>).
